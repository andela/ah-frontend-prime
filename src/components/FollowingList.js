import React from "react";

const FollowingList = props => {
  const { following } = props;
  return (
    <div className="modal-dialog" role="document">
      <div className="modal-content">
        <div className="modal-header">
          <h5 className="modal-title">My Following</h5>
          <button
            type="button"
            className="close"
            data-dismiss="modal"
            aria-label="Close"
          >
            <span aria-hidden="true">&times;</span>
          </button>
        </div>
        <div className="modal-body">
          {following && following.length > 0
            ? following.map(item => (
                <div
                  className="row"
                  id="following"
                  key={item.username}
                  onClick={() => {
                    document.location.href = `/profile/${item.username}`;
                  }}
                  key={item.username}
                >
                  <button className="col col-lg-3" style={{ border: "none" }}>
                    <img
                      src={
                        item.image
                          ? `${item.image}`
                          : "https://cdn.pixabay.com/photo/2013/07/13/10/07/man-156584__340.png"
                      }
                      alt="profile"
                      className="profile-page"
                    />
                  </button>
                  <div className="col col-lg-9">
                    <h5>{item.username}</h5>
                    <p>{item.full_name}</p>
                  </div>
                  <br />
                </div>
              ))
            : "No one is on the following List"}
        </div>
        <div class="modal-footer" />
      </div>
    </div>
  );
};

export default FollowingList;
