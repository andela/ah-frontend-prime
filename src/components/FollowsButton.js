import React from "react";

const FollowsButton = props => {
  const { isfollowing, unfollowUser, followUser } = props;
  return (
    <button
      onClick={
        isfollowing
          ? unfollowUser.bind(this, sessionStorage.getItem("userview_name"))
          : followUser.bind(this, sessionStorage.getItem("userview_name"))
      }
      className="btn btn-outline-success"
    >
      {isfollowing ? "Unfollow" : "Follow"}
    </button>
  );
};

export default FollowsButton;
