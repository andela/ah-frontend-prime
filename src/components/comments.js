import React, { Component } from "react";
import moment from "moment";
import { articleComments, fetchComments } from "../actions/commentsActions";
import { connect } from "react-redux";
import "../styles/comments.scss";


export class Comments extends Component {
  constructor(props) {
    super(props);
    this.state = {
      comments: ""
    };
  }

  componentDidMount() {
    const { fetchComments, articleSlug } = this.props;
    fetchComments(articleSlug);
  }

  handleOnChange = event => {
    event.preventDefault();
    const { name, value } = event.target;
    this.setState({ [name]: value });
  };

  handleOnSubmit = event => {
    event.preventDefault();
    const { articleSlug, articleComments, fetchComments } = this.props;
    const { comments } = this.state;
    const data = {
      body: comments
    };
    articleComments(data, articleSlug);
    fetchComments(articleSlug);
  };

  render() {
    const { commentsMessage } = this.props;
    const comment = commentsMessage.map(message => (
      <section className="comments" key={message.id}>
        <article className="comment">
          <a className="comment-img" href="#non">
            <img src={message.author.image} alt="" width="50" height="50" />
          </a>
          <div className="comment-body">
            <div className="text">
              <p>{message.body}</p>
            </div>
            <p className="attribution">
              by <a href={`/profile/${message.author.username}`}>{message.author.username}</a> at {moment(message.createdAt).format(
                          "h:mm:ss a [on] MMMM Do YYYY.")}
                        
            </p>
          </div>
        </article>
      </section>
    ));
    return (
      <div className="row">
        <div className="col col-lg-1" />
        <div className="col col-lg-10">
          <div className="row">
            <div className="col col-lg-1" />
            <div className="col col-lg-11">
              <form
                method="POST"
                onSubmit={this.handleOnSubmit}
                className="comment-forms"
              >
                <textarea
                  className="form-control"
                  name="comments"
                  value={this.state.comments}
                  onChange={this.handleOnChange}
                >
                  comments here
                </textarea>
                <button
                  type="submit"
                  className="btn btn-comment"
                  style={{ marginTop: "10px" }}
                >
                  comment
                </button>
              </form>
            </div>
          </div>
          {comment}
        </div>
        <div className="col col-lg-1" />
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  commentsSuccess: state.comments.success,
  commentsErrors: state.comments.errors,
  commentsMessage: state.getComments.comments
});
export default connect(
  mapStateToProps,
  { articleComments, fetchComments }
)(Comments);
