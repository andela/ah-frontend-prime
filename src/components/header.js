import React from "react";
import PropTypes from "prop-types";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import InputBase from "@material-ui/core/InputBase";
import { fade } from "@material-ui/core/styles/colorManipulator";
import { withStyles } from "@material-ui/core/styles";
import SearchIcon from "@material-ui/icons/Search";

const styles = theme => ({
  root: {
    width: "100%"
  },
  grow: {
    flexGrow: 1
  },
  menu: {
    marginLeft: -12,
    marginRight: 20
  },
  title: {
    display: "none",
    [theme.breakpoints.up("sm")]: {
      display: "block"
    }
  },
  inputRoot: {
    color: "inherit",
    width: "100%"
  },
  inputInput: {
    paddingTop: theme.spacing.unit,
    paddingRight: theme.spacing.unit,
    paddingBottom: theme.spacing.unit,
    paddingLeft: theme.spacing.unit * 9,
    transition: theme.transitions.create("width"),
    width: "100%",
    [theme.breakpoints.up("sm")]: {
      width: 118,
      "&:focus": {
        width: 180
      }
    }
  },
  searchIcon: {
    width: theme.spacing.unit * 8,
    height: "100%",
    position: "absolute",
    pointerEvents: "none",
    display: "flex",
    alignItems: "center",
    justifyContent: "center"
  },
  bar: {
    backgroundColor: "#003153",
    boxShadow: "none"
  },
  search: {
    position: "relative",
    borderRadius: theme.shape.borderRadius,
    backgroundColor: fade(theme.palette.common.white, 0.18),
    "&:hover": {
      backgroundColor: fade(theme.palette.common.white, 0.29)
    },
    marginLeft: 0,
    width: "100%",
    [theme.breakpoints.up("sm")]: {
      marginLeft: theme.spacing.unit,
      width: "auto"
    }
  }
});

function TopAppBar(props) {
  const { classes } = props;
  return (
    <div className={classes.root}>
      <AppBar position="static" className={classes.bar}>
        <Toolbar>
          <Typography
            className={classes.title}
            variant="h6"
            color="inherit"
            noWrap
          >
            Authors Haven
          </Typography>
          <div className={classes.grow} />

          <div className={classes.search}>
            <div className={classes.searchIcon}>
              <SearchIcon />
            </div>
            <InputBase
              placeholder="Searchâ€¦"
              classes={{
                root: classes.inputRoot,
                input: classes.inputInput
              }}
            />
          </div>
          <div className="nav-div">
            <a href="login" className="nav-links">
              Sign In
            </a>
            &nbsp;&nbsp; | &nbsp;&nbsp;
            <a href="register" className="nav-links">
              Get Started
            </a>
          </div>
        </Toolbar>
      </AppBar>
    </div>
  );
}

TopAppBar.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(styles)(TopAppBar);
