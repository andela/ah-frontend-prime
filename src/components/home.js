import React, { Component } from "react";
import { connect } from "react-redux";
import "../styles/app.scss";
import moment from "moment";
import { Link } from "react-router-dom";
import { getArticlesAction } from "../actions/getArticles";

export class Home extends Component {
  constructor(props) {
    super(props);
    this.fetchNext = this.fetchNext.bind(this);
    this.fetchPrevious = this.fetchPrevious.bind(this);
  }
  componentDidMount() {
    const { getArticlesAction } = this.props;
    getArticlesAction();
  }
  fetchNext() {
    const { next, getArticlesAction } = this.props;
    getArticlesAction(next);
  }
  fetchPrevious() {
    const { previous, getArticlesAction } = this.props;
    getArticlesAction(previous);
  }
  render() {
    const { articles } = this.props;
    const articleList = articles.length ? (
      articles.map(article => {
        return (
          <div className="article" key={article.slug}>
            <div className="inner">
              <Link
                className="links"
                to={{
                  pathname: "/article/" + article.slug
                }}
              >
                <div className="article-card">
                  <h3>{article.title.substring(0, 30)} ...</h3>
                  <p>
                    <b>{article.description.substring(0, 60)} ...</b>
                  </p>
                  <span>
                    <b>Author: {article.author.username}</b>
                  </span>
                  <br />

                  <span className="card-date">
                    Created at{" "}
                    {moment(article.createdAt).format(
                      "h:mm:ss a [on] MMMM Do YYYY."
                    )}
                  </span>
                  <br />
                  <p>
                    <b>{article.reading_time}</b>
                  </p>
                </div>

                <div>
                  <img
                    src={
                      article.image
                        ? article.image
                        : "https://previews.123rf.com/images/twindesigner/twindesigner1708/twin" +
                          "designer170800135/84202763-ah-brush-letter-logo-design-with-black-" +
                          "circle-creative-brushed-letters-icon-logo-.jpg"
                    }
                    className="img-tile"
                  />
                </div>
              </Link>
            </div>
          </div>
        );
      })
    ) : (
      <div>No Articles in Authors Haven Yet</div>
    );

    return (
      <div className="big-container">
        <div className="articles-top">
          <div className="best-article">
            <div className="inner">
              <img
                src={
                  "http://allpicts.in/download/22916/2018/03/Natural_Images_HD_1" +
                  "080p_Download_with_Waterfall_in_Tollymore_Forest_Park-1440x900.jpg/"
                }
                className="img-top"
              />
              <div className="inner">
                <div className="inner">
                  <div className="best-article-card">
                    <h3>
                      Facebook's Plan to Fuse its Messaging Apps Is Not about
                      Your Privacy Just follow the Money
                    </h3>
                    <span>
                      <b>Author: Yasha Levin in OneZero</b>
                    </span>
                    <br />
                    <span className="card-date">Created a few hours ago</span>
                  </div>
                </div>
              </div>
            </div>
            <div />
          </div>
          <div className="top-articles">{articleList}</div>
          <div className="top-articles">{articleList}</div>
        </div>

        <div className="articles">
          <div className="heading">
            <h3>Featured Articles</h3>
          </div>
          <hr />
          {articleList}
        </div>
        <div className="articles1">
          <div className="heading">
            <h3>Popular Articles</h3>
          </div>
          <hr />
          {articleList}
        </div>
        <div>
          <button
            className="btn btn-outline-success btn eian"
            onClick={() => this.fetchPrevious()}
          >
            Previous
          </button>{" "}
          <button
            className="btn btn-outline-success btn aken"
            onClick={() => this.fetchNext()}
          >
            Next
          </button>
        </div>
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  articles: state.getArticlesReducer.articles,
  next: state.getArticlesReducer.next,
  previous: state.getArticlesReducer.previous
});

export default connect(
  mapStateToProps,
  { getArticlesAction }
)(Home);
