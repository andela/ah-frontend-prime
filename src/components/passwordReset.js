import React from "react";
import "../styles/passwordreset.scss";
import Loader from "react-loader";

export const PasswordResetComponent = props => {
  const {
    password,
    passwordConfirm,
    handleOnChange,
    errors,
    handleOnSubmit,
    isLoading
  } = props;
  return (
    <div>
      <div className="container">
        <div className="row">
          <div className="col col-lg-3 col-sm-1" />
          <div className="col col-lg-6  col-md-12 col-sm-12 password-reset-container">
            <div className="">
              <h2>Password Reset</h2>
              <p>
                Please enter and confirm a new password that you will use to
                sign in into Author's haven
              </p>
              <form method="post" onSubmit={handleOnSubmit}>
                <div className="form-group">
                  <input
                    type="password"
                    id="password"
                    name="password"
                    required
                    className="form-control"
                    placeholder="Password"
                    value={password}
                    onChange={handleOnChange}
                  />
                  {errors.password.length > 0 ? (
                    <span className="form-errors-email">{errors.password}</span>
                  ) : (
                    ""
                  )}
                </div>
                <div className="form-group">
                  <input
                    type="password"
                    id="passwordConfirm"
                    name="passwordConfirm"
                    className="form-control"
                    required
                    placeholder="confirm Password"
                    value={passwordConfirm}
                    onChange={handleOnChange}
                  />
                  {errors.passwordConfirm.length > 0 ? (
                    <span className="form-errors-email">
                      {errors.passwordConfirm}
                    </span>
                  ) : (
                    ""
                  )}
                  <br />
                </div>
                <button type="submit" className="btn btn-primary">
                  Reset
                  <span>{isLoading ? <Loader /> : ""}</span>
                </button>
              </form>
            </div>
          </div>
          <div className="col col-lg-3 col-sm-1 col-sm-auto" />
        </div>
      </div>
    </div>
  );
};
export default PasswordResetComponent;
