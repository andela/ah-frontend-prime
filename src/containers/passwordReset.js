import React, { Component } from "react";
import { connect } from "react-redux";
import PasswordResetComponent from "../components/passwordReset";
import { passwordValidate } from "../utils/regex";
import { passwordReset } from "../actions/passwordResetActions";

export class PasswordResetContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      password: "",
      passwordConfirm: "",
      errors: {
        password: "",
        passwordConfirm: ""
      }
    };
  }

  componentWillReceiveProps = nextProps => {
    if (nextProps.redirectToLogin === true) {
      const { history } = this.props;
      history.push("/login");
    }
  };

  handleOnChange = event => {
    event.preventDefault();
    let errors = this.state.errors;
    const { name, value } = event.target;
    switch (name) {
      case "password":
        errors.password = passwordValidate.test(value)
          ? ""
          : "password must have a number, lowercase char,special char and min of 8 chars";
        break;
      case "passwordConfirm":
        if (value !== this.state.password) {
          errors.passwordConfirm = "your passwords don't match";
        } else {
          errors.passwordConfirm = "";
        }
        break;
      default:
        return null;
    }
    this.setState({ errors, [name]: value });
  };
  handleOnSubmit = event => {
    event.preventDefault();
    const { passwordReset } = this.props;
    const { token } = this.props.match.params;
    const { password, passwordConfirm } = this.state;
    const data = {
      password: password,
      confirmpassword: passwordConfirm
    };
    passwordReset(data, token);
  };

  render() {
    const { password, passwordConfirm, errors } = this.state;
    const { isLoading } = this.props;
    return (
      <div>
        <PasswordResetComponent
          password={password}
          passwordConfirm={passwordConfirm}
          handleOnChange={this.handleOnChange}
          handleOnSubmit={this.handleOnSubmit}
          errors={errors}
          isLoading={isLoading}
        />
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  success: state.passChange.success,
  error: state.passChange.error,
  isLoading: state.passChange.isLoading,
  redirectToLogin: state.passChange.redirectToLogin
});
export default connect(
  mapStateToProps,
  { passwordReset }
)(PasswordResetContainer);
