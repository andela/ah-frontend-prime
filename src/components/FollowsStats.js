import React from "react";
import FollowingList from "./FollowingList";
import FollowersList from "./FollowersList";

const FollowsStats = props => {
  const { following_count, followers_count, followers, following } = props;
  return (
    <div className="row">
      <div
        className="modal fade"
        id="exampleModal"
        tabIndex={-1}
        role="dialog"
        aria-labelledby="exampleModalLabel"
        aria-hidden="true"
      >
        <FollowingList following={following} />
      </div>
      <div
        className="modal fade"
        id="exampleModal1"
        tabIndex={-1}
        role="dialog"
        aria-labelledby="exampleModalLabel"
        aria-hidden="true"
      >
        <FollowersList followers={followers} />
      </div>
      <a
        className="col-md-4"
        data-toggle="modal"
        data-target="#exampleModal1"
        href="#"
      >
        {followers_count === 1 ? "Follower" : "Followers"}
        {" : "}
        <button className="stats">
          <strong>{followers_count ? followers_count : 0} </strong>
        </button>
      </a>
      <a
        className="col-md-4"
        data-toggle="modal"
        data-target="#exampleModal"
        href="#"
      >
        Following{" : "}
        <button className="stats">
          <strong>{following_count ? following_count : "0"} </strong>
        </button>
      </a>
    </div>
  );
};

export default FollowsStats;
