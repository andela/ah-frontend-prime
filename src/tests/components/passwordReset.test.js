import React from "react";
import { shallow } from "enzyme";
import { PasswordResetComponent } from "../../components/passwordReset";

let props = {
  password: "",
  passwordConfirm: "",
  handleOnChange: jest.fn(),
  errors: {
    password: "",
    passwordConfirm: ""
  },
  handleOnSubmit: jest.fn()
};
let instance;
describe("passwordComponent", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<PasswordResetComponent {...props} />);
    instance = wrapper.instance();
  });
  it("should render properly", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should display password reset errors ", () => {
    wrapper.setProps({
      errors: {
        password: "12345@My",
        passwordConfirm: "12345@My"
      }
    });
    expect(wrapper).toMatchSnapshot();
  });
  it("should show loaders ", () => {
    wrapper.setProps({ isLoading: "<Loader />" });
    expect(wrapper).toMatchSnapshot();
  });
});
