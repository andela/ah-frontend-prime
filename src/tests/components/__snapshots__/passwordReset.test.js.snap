// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`passwordComponent should display password reset errors  1`] = `ShallowWrapper {}`;

exports[`passwordComponent should render properly 1`] = `ShallowWrapper {}`;

exports[`passwordComponent should show loaders  1`] = `ShallowWrapper {}`;
